XDS_PROTOCOL.md

```diff
@@ -0,0 +1,216 @@
+# xDS REST and gRPC protocol
+
+Envoy discovers its various dynamic xDS resources via the filesystem or by querying
+one or more management servers. Resources are requested via _subscriptions_, by
+either specifying a filesystem path to watch, initiating gRPC streams or
+REST-JSON polling. The latter two methods involve sending requests with a
+[`DiscoveryRequest`](api/discovery.proto#L24) proto payload. Resources are
```

Comment:
nit: best practice for code line links like this is to link to a SHA which won't move. (Same below).